helper-cli/src/main/kotlin/commands/packageconfig/CreateCommand.kt

```diff
@@ -84,21 +84,24 @@ internal class CreateCommand : CliktCommand(
         val scanResultsStorage = FileBasedStorage(LocalFileStorage(scanResultsStorageDir))
         val scanResults = scanResultsStorage.read(packageId).getOrDefault(emptyList())
 
-        scanResults.find { it.provenance is RepositoryProvenance }?.provenance
-            ?.writePackageConfigurationFile("vcs.yml")
-        scanResults.find { it.provenance is ArtifactProvenance }?.provenance
-            ?.writePackageConfigurationFile("source-artifact.yml")
+        scanResults.find { it.provenance is RepositoryProvenance }?.let {
+            createPackageConfiguration(it.provenance)
+        }?.writeToFile()
+
+        scanResults.find { it.provenance is ArtifactProvenance }?.let {
+            createPackageConfiguration(it.provenance)
+        }?.writeToFile()
     }
 
-    private fun Provenance.writePackageConfigurationFile(filename: String) {
-        val packageConfiguration = createPackageConfiguration(this)
+    private fun PackageConfiguration.writeToFile() {
+        val filename = "vcs.yml".takeIf { vcs != null } ?: "source-artifact.yml"
```

Comment:
I find `val filename = if (vcs != null) "vcs.yml" else "source-artifact.yml"` easier to read, and it's also shorter.